### ISSUE TITLE
Drag Tag to add it

### ISSUE THREAD

**Issue by ybabel** [Jermolene/TiddlyWiki5#1548]
Hello,

what about dragging a Tag from the side view to add it to  a Tiddler ?

**Comment by ybabel** [comment]
Ideally, I have in mind something like the wonderful "Kanban" (TWc) plugin : http://kanban.tiddlyspot.com/

**Comment by twMat** [comment]
>what about dragging a Tag from the side view to add it to a Tiddler ?

Makes sense particularly if you have tab *Tags* open. More generally, I think it is an interesting idea to drag-drop *any* internal tiddler title to create a link to it. Perhaps dropping it on specific fields in a tiddler could turn it into content for that field - i.e dropping it onto the tags area/field turns the title into a tag, whereas dropping it on the text area creates a text link, etc. I believe a generally more extensive drag-and-drop handling is intended to come, including the dragging of links (albeit primarily for extended importing between TW's, I think). 

The Kanban plugin you refer to, in turn refers to (and is build on) http://dragsort.codeplex.com/ which appearently is still maintained and updated. Could this open source project perhaps be of value for our intended drag-n drop features - @Jermolene ?

**Comment by ybabel** [comment]
Precisely !
I think there is a huge potential in D&D in TW5 : tags, links, and maybe automatic transclusion : droping a field for example or a DataTiddler could create a transclusion.
I didn't think about all the possibilities... they are maybe endless.

For example we could imagine that if we drag&drop part of the content of a Tiddler : 
* create a new tiddler with the content droped
* replace the part of the previous tiddler with a transclusion of the new tiddler

that would be the basis of a refactoring mecanism ...

**Comment by twMat** [comment]
>if we drag&drop part of the content of a Tiddler :
> *   create a new tiddler with the content droped

Note that this is already possible but via the importing mechanism.

>replace the part of the previous tiddler with a transclusion of the new tiddler

This woudl be very useful! I think there was some talk that for the "visual edit bar" (I'm forgetting the correct name right now) there might be a "snip out and replace with transclusion" like there was in Erics TWc version. But to have it replaced with a transclusion from *dragging* feels even better and is a new idea, I think.

Further idea; drop the text portion either onto a tiddler *or* onto a mere tiddler link and have it append to the text in that tiddler. I.e so you wouldn't have to have it open.

**Comment by Spangenhelm** [comment]
:+1: endless possibilities indeed but there is now more than 300 opened issues as we speak so unless jeremy sets up a top priority on this it will take forever to see it coming...

**Comment by Innomen** [comment]
As a new non-coder user I just want to say like the idea of drag and drop options. I'm here today expressly because I can't drag the new tiddler button to my bookmarks. I would like it to be a single click operation. (As I can do with wordpress posts, click new, start writing.)

I would also like the ability to have a bookmark a tiddler template, like imagine a save link tiddler where clicking it would create a new tiddler with a tag preselected the date inserted and the link text. A bit like how sharing buttons work in social media plugins and bookmarklets.

**Comment by Spangenhelm** [comment]
Hi @Innomen, i don't think you can bookmark the button itself but only a tiddler that contains the new tiddler button. (via the permalink)
For creating a new tiddler with tag date and text maybe check the "tm-new-tiddler" widgetmessage.

Hope this can help..
+

**Comment by pmario** [comment]
@Innomen Hi,
You should create a new topic with your suggestions. IMO they will get lost here, since the OP is a different topic.

**Comment by Innomen** [comment]
@pmario 
Will do, thank you :) Sorry for going off topic also I suppose hehe.

**Comment by pmario** [comment]
With the discussions here, it's easy to be OT. So there is no problem, if we find back :) I've seen the new issue #1612. And there seems to be some valuable discussion too ;)

**Comment by pmario** [comment]
@ybabel ... please test the new Drag and Drop mechanism in v5.1.14, ... There is no kanban board implemented, but imo it's now easy to create one. 

imo this issue can be closed.

**Comment by tobibeer** [comment]
There might be a dropzone in the tiddler viewtemplate that when hovering with a tiddler you can drop it there to turn it into a tag (insofar as it is not one yet).

**Comment by tobibeer** [comment]
On the other hand, I am a minimalist, I think. So, the idea of using an alternative workaround for adding tags... to me, it only introduces complexity and future complications.

**Comment by pmario** [comment]
@ybabel ... IMO there are 2 kanban implementations now: 

Dropboard:  https://groups.google.com/forum/#!searchin/tiddlywiki/kanban%7Csort:date/tiddlywiki/0YDWlU0Ls4Y/laLu5KMRBAAJ and a second one 

Tekan: https://ibnishak.github.io/Tesseract/projects/tekan/Tekan.html
